feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,321 @@
+		t.Logf("LLDP ChassisId got: %s", state.GetChassisId())
+	} else {
+		t.Errorf("LLDP chassisID is not proper, got %s", state.GetChassisId())
+	}
+	if state.GetChassisIdType() != 0 {
+		t.Logf("LLDP ChassisIdType got: %s", state.GetChassisIdType())
+	} else {
+		t.Errorf("LLDP chassisIdType is not proper, got %s", state.GetChassisIdType())
+	}
+	if state.GetSystemName() != "" {
+		t.Logf("LLDP SystemName got: %s", state.GetSystemName())
+	} else {
+		t.Errorf("LLDP SystemName is not proper, got %s", state.GetSystemName())
+	}
+
+	got := state.GetInterface(port.Name()).GetName()
+	want := conf.GetInterface(port.Name()).GetName()
+
+	if lldpEnabled && got != want {
+		t.Errorf("LLDP interfaces/interface/state/name = %s, want %s", got, want)
+	}
+}
+
+// checkOtgLLDPMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
+func checkOtgLLDPMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
+	for _, lldp := range c.Lldp().Items() {
+		lastValue, ok := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameOut().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			txPackets, _ := v.Val()
+			return v.IsPresent() && txPackets != 0
+		}).Await(t)
+		if !ok {
+			txPackets, _ := lastValue.Val()
+			t.Errorf("LLDP sent packets got: %v, want: > 0.", txPackets)
+		}
+		framesIn, _ := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameIn().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			return v.IsPresent()
+		}).Await(t)
+		otgutils.LogLLDPMetrics(t, otg, c)
+		if lldpEnabled {
+			if rxPackets, _ := framesIn.Val(); rxPackets == 0 {
+				t.Errorf("LLDP received packets got: %v, want: > 0.", rxPackets)
+			}
+		} else {
+			if rxPackets, _ := framesIn.Val(); rxPackets != 0 {
+				t.Errorf("LLDP received packets got: %v, want: 0.", rxPackets)
+			}
+		}
+	}
+}
+
+// checkOtgLLDPNeighbors verifies OTG side lldp neighbor states
+func checkOtgLLDPNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) {
+	otgutils.LogLLDPNeighborStates(t, otg, c)
+
+	for lldp, lldpNeighbors := range expLldpNeighbors {
+		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldp).State())
+
+		if len(lldpNeighbors) == 0 {
+			if lldpState.LldpNeighborDatabase != nil {
+				t.Errorf("LldpNeighborDatabase = %v; want nil", lldpState.LldpNeighborDatabase)
+				return
+			}
+		} else {
+			neighbors := lldpState.GetLldpNeighborDatabase().LldpNeighbor
+			for _, lldpNeighbor := range lldpNeighbors {
+				neighborFound := false
+				for _, neighbor := range neighbors {
+					if neighbor.GetChassisId() == lldpNeighbor.chassisId &&
+						neighbor.GetChassisIdType() == lldpNeighbor.chassisIdType &&
+						neighbor.GetPortId() == lldpNeighbor.portId &&
+						neighbor.GetPortIdType() == lldpNeighbor.portIdType &&
+						neighbor.GetSystemName() == lldpNeighbor.systemName {
```

Comment:
I suggest factoring this out to a method of lldpNeighbors.

```
func (ln *lldpNeighbors) Equal(...) bool {
}
```

Here just:

```
if lldpNeighbor.Equal(neighbor) {
  neighborFound = true
  break
}
```

Reply:
Added the Equal method to the lldpNeighbours type